### PR TITLE
Script for adding missing case email attachments

### PR DESCRIPTION
**Pull request by mikeylockwood**

### CODE REVIEW

File: data_migration/management/commands/add_missing_constabulary_email_attachments.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        case_email = CaseEmail.objects.get(pk=dm_case_email.pk)
+        app = case_email.export_applications.first().get_specific_model()
+
+        dm_file_pks = dm.File.objects.filter(target_id=row["TARGET_ID"]).values_list(
+            "pk", flat=True
+        )
+        files = app.supporting_documents.filter(pk__in=dm_file_pks)
+
+        if self.dry_run:
+            print(f"{app.reference} - Adding {files.count()} files to BEIS Email")
```
Comment (marcuspp) [inline data_migration/management/commands/add_missing_constabulary_email_attachments.py:125]: self.stdout.write

File: data_migration/management/commands/add_missing_constabulary_email_attachments.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                        row = next(rows)
+                        self.stdout.write(f"Processing file: {row['FILE_TARGET_ID']}")
+                        created = self.add_file_to_db(row)
+                        if created:
+                            self.upload_file_to_s3(row)
+                except StopIteration:
+                    pass
+
+    def upload_file_to_s3(self, row: dict[str, Any]) -> None:
+        """Uploads fir attachment file to S3."""
```
Comment (marcuspp) [inline data_migration/management/commands/add_missing_constabulary_email_attachments.py:160]: not firs

File: data_migration/management/commands/add_missing_constabulary_email_attachments.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            action="store_true",
+        )
+
+    def get_db(self, query_name: str) -> OracleDBProcessor:
+        """Returns a DB processor that facilitates executing database queries on the v1 oracle database."""
+        return OracleDBProcessor(100, [query_name], 100)
+
+    def handle(self, *args: Any, **options: Any) -> None:
+        self.dry_run = options["dry_run"]
+        self.add_files_constabulary_email_attachments()
```
Comment (marcuspp) [inline data_migration/management/commands/add_missing_constabulary_email_attachments.py:90]: Think this needs to call `add_beis_email_attachments`

**Review by marcuspp — COMMENTED** [review]
just had a quick look over things

File: data_migration/management/commands/add_missing_constabulary_email_attachments.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        if row["FILE_SIZE"] and int(row["FILE_SIZE"]) > file_size_limit:
+            s3_web.upload_file_obj_to_s3_in_parts(row["BLOB_DATA"], row["PATH"])
+        else:
+            s3_web.upload_file_obj_to_s3(row["BLOB_DATA"], row["PATH"])
+
+    def add_file_to_db(self, row: dict[str, Any]) -> bool:
+        """Finds the relevant case email in the V2 database add adds the file attachment data."""
+        file_data = dict(
+            filename=row["FILENAME"],
+            content_type=row["CONTENT_TYPE"],
```
Comment (marcuspp) [inline data_migration/management/commands/add_missing_constabulary_email_attachments.py:173]: Do all files have a content type set?

Comment (mikeylockwood) [reply data_migration/management/commands/add_missing_constabulary_email_attachments.py:173]: I checked and they do

**Review by marcuspp — APPROVED** [review]

**Pull request merged**